nifi-nar-bundles/nifi-amqp-bundle/nifi-amqp-processors/src/main/java/org/apache/nifi/amqp/processors/ConsumeAMQP.java

```diff
@@ -159,8 +158,8 @@ protected void processResource(final Connection connection, final AMQPConsumer c
         }
 
         if (lastReceived != null) {
-            session.commit();
-            consumer.acknowledge(lastReceived);
+            final GetResponse finalGetResponse = lastReceived;
+            session.commitAsync(() -> consumer.acknowledge(finalGetResponse), null);
```

Comment:
Because it didn't exist when I wrote that line of code :) After I passed `null` 3-4 times I realized that it would be `null` more often than not so added the override. Will fix.